packages/sol6/src/yang/etsi-nfv-ns.yang

```diff
@@ -148,31 +154,61 @@ submodule etsi-nfv-ns {
           }
           description
             "References the descriptor of the NS VL instance to
-             which the SAP instantiated from this SAPD connects to.";
+            which the SAP instantiated from this SAPD connects to.";
           reference
             "GS NFV IFA014: Section 6.2.3.2 Sapd information element";
         }
 
-	choice associated-cpd-id {
-	  leaf vnf-ext-cpd {
-	    type leafref {
-	      path "deref(../virtual-link-desc)/../../../vnfd/ext-cpd/id";
-	    }
-	  }
-	  leaf pnf-ext-cpd {
-	    type leafref {
-	      path "deref(../virtual-link-desc)/../../../pnfd/ext-cpd/id";
-	    }
-	  }
-	  leaf sapd {
-	    type leafref {
-	      path "../../id";
-	    }
-	  }
-	}
-
-	uses common:cpd;
+        choice associated-cpd-id {
+            container vnf {
+              leaf vnfd-id {
+                mandatory true;
+                type leafref {
+                   path "/nfv:nfv/nfv:vnfd/nfv:id";
```

Comment:
Do we need the prefix nfv in the path statement? Particularly since this is a submodule of nfv?

Reply:
We do prefer to keep the prefixes... for leafrefs it shouldn't matter, but for XPATH expressions you can get weird results. 

I'm open to suggestions, but nowadays we always prefix everything...